shinyTools 25.01.0: Feat/35 user points

arunge opened the pull request:
# shinyTools 25.01.0

## New Features
- new module (`customPoints`) that allows a user to add points to a plot and format them (#35)

## Updates
- _headerUI_: keep only logos for _Pandora_ and _Isomemo_, add button to _Data Search_ app (#1)

-----

|Add custom points | Modify the layout |
|---|---|
|![image](https://github.com/user-attachments/assets/7d343c6b-88ca-4164-88f3-b6b13e49be49)|![image](https://github.com/user-attachments/assets/f223f292-b9ff-40f0-8e6d-bf895063f293)|

arunge commented:
@f-lukas Could you please test the new module "customPoints"? There is a test app at the button of the file `03-customPoints.R`. Let me know if you have questions or issues! Thanks! :pray:

f-lukas commented:
I tested it with the example app. In general everything works as it should. I have only noted two things:

- The user selects a value for the point and can also specfiy a minimum and a maximum. At first I thought everything that is generated (point and line) is considered as the point. In the meantime I have found out that changing only the point is possible via the "point" tab and changing the line is possible via the "error" tab. I am totally fine with having this in different tabs. The only thing I would suggest is to add more description to the "add" tab. E.g. instead of "minimum" and "maximum" it would be very helpful to have a little description e.g. optinally add minimum and maximum that creates a line through the specified point. The layout of the line can be changed in the error tab. Another option could be to rename the error tab to "line (error)" because I was confused that error is refering to the line.
- The second thing is, if a value and min and max is selected for both x and y axis the result looks like on the screenshot. Is it correct that there is a line at the upper end and lower end? E.g. when selecting the values below, I would expect a point at x=4 & y=20 and also two lines that go through that point but I would not expect lines at the upper and lower end.

![Bildschirmfoto vom 2025-01-22 13-54-17](https://github.com/user-attachments/assets/77d6b8f0-d4e0-49c5-97b4-5d69792e2bba)
![Bildschirmfoto vom 2025-01-22 14-07-39](https://github.com/user-attachments/assets/e0108623-96b4-4d37-a808-5b25281ac1f8)

arunge commented:
@f-lukas Thanks for your feedback, both are good points. Please, let me know if the points are solved now:

1. To make the labels more precise, I added a help text and updated the labels as follows. Do you think this is clearlier now?
![image](https://github.com/user-attachments/assets/7d343c6b-88ca-4164-88f3-b6b13e49be49)

2. The additional lines are meant to be the whiskers/caps of the error bars. They are drawn per default by `ggplot2::geom_bar()` and `ggplot2::geombarh()`. However, finding default values here is too difficult without knowing the x- and y-axis ranges (which is the case inside the module). Now, I set the default value to "0" (as displayed for "Point 1"). However, the user may still add caps if needed (as displayed for "Point 2").
![image](https://github.com/user-attachments/assets/f223f292-b9ff-40f0-8e6d-bf895063f293)

f-lukas reviewed (approved):
@arunge thank you. The helptext and new names are indeed very helpful :+1: 

> The additional lines are meant to be the whiskers/caps of the error bars.

Okay, now it makes sense. I think your change is a very good idea.

Feel free to merge once the pipeline runs without errors.
